tangelo/linq/target/target_sympy.py

```diff
@@ -75,8 +75,10 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
         frequencies = dict()
         for vec, prob in measurements:
             prob = simplify(prob, tolerance=1e-4)
-            bistring = "".join(str(bit) for bit in reversed(vec.qubit_values))
-            frequencies[bistring] = prob
+            prob = nsimplify(prob, tolerance=1e-4, rational=True)
```

Comment:
This line is what enables us to have rational approximations of floats up to a given precision, right ?
Out of curiosity: do you know how time-consuming is it and how helpful do you think this is in practice ?

I was naively expecting fractions to be exact in the output expressions.

(commenting this line on 1-2 use cases of decent size will tell you how much time it takes)

Reply:
It is, but I found out it is not simpler to leave fractions: in my use-cases, I was getting exact fractions in the form of `3290239049023/348923890208423094823` or something uglier like that. I prefer getting `9.42967547e-9` than the fraction in this case.

In the case of speed, it will need investigation. This is also why I set a loose tolerance, so it does not look for a machine-precision exact fraction.

Reply:
Sorry, what I meant is, is there value in request a fraction that is an approximation of a floating point number, rather than requesting the floats, up to a level of precision ?
(e.g `0.71428` instead of `5/7` etc when we know 5/7 is actually not exact and can't be directly compared to machine calculations, stored in floats)

(for small numbers, such as your example, scientific notation is nice unless we proclaim things below a given threshold to be 0)